include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -603,11 +617,14 @@ __parallel_radix_sort_iteration(_ExecutionPolicy&& __exec, ::std::size_t __segme
 
     // 2. Scan Phase
     sycl::event __scan_event = __radix_sort_scan_submit<_RadixLocalScanKernel, _RadixGlobalScanKernel, __radix_bits>(
-        __exec, __scan_wg_size, __segments, __tmp_buf, __count_event);
+        ::std::forward<_ExecutionPolicy>(__exec), __scan_wg_size, __segments, ::std::forward<_TmpBuf>(__tmp_buf),
+        __count_event);
 
     // 3. Reorder Phase
     sycl::event __reorder_event = __radix_sort_reorder_submit<_RadixReorderKernel, __radix_bits, __is_comp_asc>(
-        __exec, __segments, __block_size, __reorder_sg_size, __radix_iter, __in_rng, __out_rng, __tmp_buf, __scan_event
+        ::std::forward<_ExecutionPolicy>(__exec), __segments, __block_size, __reorder_sg_size, __radix_iter,
+        ::std::forward<_InRange>(__in_rng), ::std::forward<_OutRange>(__out_rng), ::std::forward<_TmpBuf>(__tmp_buf),
```

Comment:
You can remove ::std::forward<_TmpBuf>(__tmp_buf) to just __tmp_buf and add the corresponding changes to __parallel_radix_sort_iteration API (const & instead of &&).

Reply:
Since we create tmp_buf and therefore have full control over it, then it is sufficient to use lvalue reference. We change `tmp_buf`, so `const` should not be used. As for `__out_rng`, I left it as rvalue reference because we swap it with `__in_rng` in odd iteration.